### PR TITLE
refactor: Convert Avatar stories to csf format

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Updated Avatar Base's const, stories, and test files.
- Updated Avatar Account's const, stories, and test files.
- Updated Avatar Favicon's const, stories, and test files.
- Updated Avatar Icon's const, stories, and test files.
- Updated Avatar Network's const, stories, and test files.
- Updated Avatar Token's const, stories, and test files.

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/8112138/a65d7fe3-295b-45b1-8a3e-6b15ea71cdaf

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Avatars/Avatar/Avatar.stories.tsx [hunk 1]
```diff
@@ -1,83 +1,73 @@
...
+            {...SAMPLE_AVATARNETWORK_PROPS}
+          />
+        );
+      case AvatarVariant.Token:
+        return (
+          <AvatarComponent
+            variant={AvatarVariant.Token}
+            {...SAMPLE_AVATARTOKEN_PROPS}
+          />
+        );
```
Comment (georgewrmarshall) [inline app/component-library/components/Avatars/Avatar/Avatar.stories.tsx:68]: This option seems to be broken


https://github.com/MetaMask/metamask-mobile/assets/8112138/0c662962-4831-4602-bc77-6d5cca9d72b5

File: app/component-library/components/Avatars/Avatar/variants/AvatarIcon/AvatarIcon.stories.tsx [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
+      options: IconColor,
+      control: {
+        type: 'select',
+      },
+      defaultValue: SAMPLE_AVATARICON_PROPS.iconColor,
+    },
+    backgroundColor: {
+      control: {
+        type: 'color',
+      },
```
Comment (georgewrmarshall) [inline app/component-library/components/Avatars/Avatar/variants/AvatarIcon/AvatarIcon.stories.tsx:38]: This should be limited to our color tokens?

Screenshot 2023-10-23 at 2 03 13 PM

File: app/component-library/components/Avatars/Avatar/variants/AvatarToken/AvatarToken.stories.tsx [hunk 3]
```diff
@@ -1,70 +1,33 @@
...
 };
+export default AvatarTokenMeta;
 
-const AvatarTokenStory = () => <AvatarToken {...getAvatarTokenStoryProps()} />;
-
-export default AvatarTokenStory;
+export const AvatarToken = {
+  args: {
+    imageSource: SAMPLE_AVATARTOKEN_PROPS.imageSource,
+    isIpfsGatewayCheckBypassed: true,
```
Comment (georgewrmarshall) [inline app/component-library/components/Avatars/Avatar/variants/AvatarToken/AvatarToken.stories.tsx:31]: `isIpfsGatewayCheckBypassed` breaks when toggled off


https://github.com/MetaMask/metamask-mobile/assets/8112138/a50bcd99-c524-4e00-bb5c-d12a9a7c2457

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Left some comments. This PR is quite large and addresses several issues that could potentially be split into multiple PRs. To make it easier on the reviewer I would suggest breaking up PRs like this in future. Pulled branch and checked
- for any remaining references to `AvatarVariants` to ensure none were overlooked.
- stories, controls and console

File: app/component-library/components/Avatars/Avatar/variants/AvatarNetwork/README.md [hunk 4]
```diff
@@ -25,10 +25,6 @@ Optional network image from either a local or remote source.
 ## Usage
 
 ```javascript
-// Replace import with relative path.
-import AvatarNetwork from 'app/component-library/components/Avatars/AvatarNetwork';
-import { AvatarSize } from 'app/component-library/components/Avatars/Avatar/Avatar';
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Avatars/Avatar/variants/AvatarNetwork/README.md:30]: importing paths will be added in again when we can do absolute import

File: app/component-library/components/Avatars/Avatar/variants/AvatarIcon/AvatarIcon.stories.tsx (hunk 2 above)
Comment (brianacnguyen) [reply app/component-library/components/Avatars/Avatar/variants/AvatarIcon/AvatarIcon.stories.tsx:38]: that means we'll need to create a flat list of our color tokens, which we dont have yet

File: app/component-library/components/Avatars/Avatar/Avatar.types.ts [hunk 5]
```diff
@@ -19,7 +23,7 @@ export enum AvatarSize {
 /**
  * Avatar variants.
  */
-export enum AvatarVariants {
+export enum AvatarVariant {
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/components/Avatars/Avatar/Avatar.types.ts:26]: done https://github.com/MetaMask/metamask-mobile/pull/7746/files

File: app/components/Views/Settings/SecuritySettings/__snapshots__/SecuritySettings.test.tsx.snap [hunk 6]
```diff
@@ -1615,6 +1615,7 @@ exports[`SecuritySettings should render correctly 1`] = `
                   "width": 32,
                 }
               }
+              testID="cellbase-avatar"
```
Comment (georgewrmarshall) [inline app/components/Views/Settings/SecuritySettings/__snapshots__/SecuritySettings.test.tsx.snap:1618]: Should these not be different at each implementation?

Comment (brianacnguyen) [reply app/components/Views/Settings/SecuritySettings/__snapshots__/SecuritySettings.test.tsx.snap:1618]: this is the result of the new testing library

**Review by georgewrmarshall — COMMENTED** [review]
Running into some issues 
- Avatar variant favicon is square is that intentional?
- Error on Avatar variant token


https://github.com/MetaMask/metamask-mobile/assets/8112138/8156a2a9-a7c8-41ae-badc-2b3e17252a82

**Comment by brianacnguyen** [comment]
> Running into some issues
> 
> * Avatar variant favicon is square is that intentional?
> * Error on Avatar variant token
> 
>  Screen.Recording.2023-11-10.at.11.49.32.AM.mov

fixed

**Review by georgewrmarshall — COMMENTED** [review]
Stories are working but looks like you have some failing tests. This is the video I would like to see as a reviewer to see that you have also checked all the stories. Keep in mind the more stories you add the more that need to be QA'd I would expect these to be checked by the author first. This will also help for faster review


https://github.com/MetaMask/metamask-mobile/assets/8112138/a65d7fe3-295b-45b1-8a3e-6b15ea71cdaf

**Review by georgewrmarshall — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**